x-pack/plugins/reporting/server/lib/export_types_registry.ts

```diff
@@ -90,16 +71,10 @@ export class ExportTypesRegistry {
 
 export function getExportTypesRegistry(): ExportTypesRegistry {
   const registry = new ExportTypesRegistry();
-
-  /* this replaces the previously async method of registering export types,
-   * where this would run a directory scan and types would be registered via
-   * discovery */
-  const getTypeFns: Array<() => ExportTypeDefinition<any, any, any, any>> = [
-    getTypeCsv,
-    getTypeCsvFromSavedObject,
-    getTypePng,
-    getTypePrintablePdf,
-  ];
+  const getTypeFns: Array<() => ExportTypeDefinition<
+    CreateJobFn<any>, // diferent type of params are not assignable to each other
+    any // ImmediateExecuteFn is not assignable to RunTaskFn<any>
+  >> = [getTypeCsv, getTypeCsvFromSavedObject, getTypePng, getTypePrintablePdf];
```

Comment:
I think we could come up with a really good solution of "immediate" export type not fitting into the model by just instantiating a new export type registry instance that can register immediate types.

An export type definition of the 2nd registry would not need a "create job" function. It can execute the job with the user's request.